src/node_contextify.cc

```diff
@@ -1200,11 +1203,38 @@ static void WatchdogHasPendingSigint(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+static void MeasureMemory(const FunctionCallbackInfo<Value>& args) {
+  CHECK(args[0]->IsInt32());
+  int32_t mode = args[0].As<v8::Int32>()->Value();
+  Isolate* isolate = args.GetIsolate();
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context;
+  if (args[1]->IsUndefined()) {
+    context = isolate->GetCurrentContext();
+  } else {
+    ContextifyContext* sandbox =
+        ContextifyContext::ContextFromContextifiedSandbox(env,
+                                                          args[1].As<Object>());
+    CHECK_NOT_NULL(sandbox);
+    context = sandbox->context();
+    if (context.IsEmpty()) {  // Not yet fully initilaized
+      return;
```

Comment:
Shouldn't you create and return a promise here?

Reply:
Alternatively, check the return value in JavaScript and return a `Promise.resolve()` from the JS function if return value is undefined. Or should it be a `Promise.reject()`? It's not clear if this is an error condition or just a "not yet available"

Reply:
I guess a rejection makes sense here (but returning from the JS land would be preferable as we could instantiate errors more easily there)

Reply:
Updated to return a rejection here.